pkg/config/config.go

```diff
@@ -372,7 +373,13 @@ func initConfig(config Config) {
 	// Note that trace-agent environment variables are parsed in pkg/trace/config/env.go
 	// since some of them require custom parsing algorithms. DO NOT add environment variable
 	// bindings here, add them there instead.
-	config.BindEnvAndSetDefault("apm_config.enabled", true)
+	if runtime.GOARCH == "386" {
+		// on Windows-32 bit, the trace agent isn't installed.  Set the default to disabled
+		// so that there aren't messages in the log about failing to start.
+		config.BindEnvAndSetDefault("apm_config.enabled", false)
```

Comment:
This is a hack that is getting us into trouble because of multiple reasons:
* Simply because the architecture is `386` it doesn't mean we're on Windows. We have GOOS for that.
* If the trace-agent isn't installed on Windows 32-bit, then the Windows installer should know about this at the place where it fails to start, not here.

This is a very out of place change. Can we move it elsewhere? Maybe to where it belongs?

Reply:
I hope you were not debugging why a 32 bit of the Agent didn't run trace-agent when you found this... it feels like something very painful to debug to finally find this 🤦‍♀. So sorry if it was the case 🙇

It's true that not checking if this is Windows was an oversight (although we didn't know we would be supporting 32 bit anywhere else, to be honest). We could fix it just by checking both GOARCH and GOOS on the same line (I can do the PR right away), but you seem to suggest a better place for this check. What would that place be?

Reply:
PR: https://github.com/DataDog/datadog-agent/pull/4881